tribits/core/package_arch/TribitsGitRepoVersionInfo.cmake

```diff
@@ -44,6 +44,74 @@
 ################################################################################
 
 
+# @FUNCTION: tribits_git_repo_sha1()
+#
+# Get the Git repo version SHA1.
+#
+# Usage::
+#
+#   tribits_git_repo_sha1(<gitRepoDir> <gitRepoSha1Out>
+#     FAILURE_MESSAGE_OUT <failureMsgOut>)
+#
+# If ``<failureMsgOut>`` is non-empty on input, then the variable
+# ``<failureMsgOut>`` will be set to a non-empty value on output with the
+# error message if an error occurs and the SHA1 for ``<gitRepoDir>`` can not
+# be computed.  If ``<failureMsgOut>`` is empty in input, and there is an
+# error, a ``FATAL_ERROR`` will be raised.
+#
+# NOTE: To get the SHA1 for the Git repo ``<gitRepoDir>``, it must contain the
+# ``.git/`` directory and the ``git`` executable var ``GIT_EXECUTABLE`` must
+# be set.  Otherwise, it is an error.
+#
+function(tribits_git_repo_sha1  gitRepoDir  gitRepoSha1Out)
+
+  cmake_parse_arguments( PARSE_ARGV 2
+    PARSE "" "" # prefix, options, one_value_keywords
+    "FAILURE_MESSAGE_OUT" # multi_value_keywords
+    )
+  tribits_check_for_unparsed_arguments()
+  tribits_assert_parse_arg_zero_or_one_value(PARSE  FAILURE_MESSAGE_OUT)
+
+  set(failureMsg "")
+
+  if (NOT GIT_EXECUTABLE)
+    set(failureMsg "ERROR: The program '${GIT_NAME}' could not be found!")
+  elseif (NOT IS_DIRECTORY "${gitRepoDir}/.git")
+    set(failureMsg "ERROR: The directory ${gitRepoDir}/.git does not exist!")
+  endif()
+
+  set(gitRepoSha1 "")
+  if (failureMsg STREQUAL "")
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} log -1 --pretty=format:"%H"
```

Comment:
Thanks!  This was a copy and paste from the function `tribits_generate_single_repo_version_string()` that actually needs those quotes because there are spaces in the `--pretty=format:<value>`.  (In that case `--pretty=format:"%h [%ad] <%ae>"`).  I verified that I can remove the spaces (and there is a strong test that verifies the code still works as it should).

Reply:
I see.

CMake has three types of arguments: unquoted, quoted, and bracketed. If the argument starts with `"`, it's a quoted argument. If not, it's an unquoted argument, even if it contains a quote later on. So, for `tribits_generate_single_repo_version_string()`, you could change it to `"--pretty=format:%h [%ad] <%ae>"` and not have to strip off the quotes afterwards.

Reply:
I am pretty sure I tried that and it did not work.  I can try it again.

Reply:
I just tried the following locally and it worked fine:

```cmake
execute_process(COMMAND git log -1 "--pretty=format:%h [%ad] <%ae>")
```

Reply:
Okay, that does seem to work correctly.  There are strong tests that break of you don't produce this line correctly.  I will include this change in the follow up PR.

Reply:
Addressed in commit https://github.com/TriBITSPub/TriBITS/pull/486/commits/4bd215a374ce28779416e62f85c7b9f9ec9976f8 in new PR #486